vault/client.go

```diff
@@ -171,7 +176,16 @@ func (b *BankVaultClient) removeSecret(removeSecret VaultSecret) error {
 	}
 
 	for _, secret := range secrets {
-
+		if isDirectory(secret) {
+			err := b.removeSecret(VaultSecret{
+				Path:  path.Join(removeSecret.Path, secret),
+				Value: "",
```

Comment:
I would omit setting the value to an empty string here. It can be confusing on the first read and it's de default anyway.